WooCommerce/Classes/ViewRelated/Dashboard/Blaze/BlazeCampaignDashboardViewModel.swift

```diff
@@ -118,25 +120,39 @@ final class BlazeCampaignDashboardViewModel: ObservableObject {
         configureResultsController()
     }
 
+    @MainActor
+    func checkAvailability() async {
+        isSiteEligibleForBlaze = await blazeEligibilityChecker.isSiteEligible()
+        try? await synchronizePublishedProducts()
+        updateAvailability()
+    }
+
     @MainActor
     func reload() async {
+        syncingError = nil
         update(state: .loading)
-        isSiteEligibleForBlaze = await blazeEligibilityChecker.isSiteEligible()
+
+        analytics.track(event: .DynamicDashboard.cardLoadingStarted(type: .blaze))
 
         guard isSiteEligibleForBlaze else {
             update(state: .empty)
```

Comment:
`isSiteEligibleForBlaze` has default value set as `false`. If we call reload without calling `checkAvailability` we will end up not checking site eligibility. 

One solution I can think of is to make `isSiteEligibleForBlaze` optional and check site eligibility from `reload` if `isSiteEligibleForBlaze` is nil. 

What do you think?

Reply:
For simplicity, I added a check for site eligibility at the beginning of `reloadData` in a0f2237bc6adabd3d29363272cdb8836377a323e. Keeping `isSiteEligibleForBlaze` optional would make it difficult to work with in other methods.